nasa/harmony
# Harmony 1894 - add support for labeling jobs

## Jira Issue ID
HARMONY-1894

## Description
Adds a new parameter `label` that can be provided with Harmony requests (OGC Coverages/EDR) to specify one ore more labels for a job.

## Local Test Steps
1. Check out this branch
2. Run the migrations or run `bin/bootstrap-harmony` which will run the migrations for you
3. Include one or more labels with a Harmony request
For Coverages:
```
curl -Ln -bj "http://localhost:3000/C1233800302-EEDTEST/ogc-api-coverages/1.0.0/collections/red_var,blue_var,green_var/coverage/rangeset/?subset=lat(20%3A60)&subset=lon(-140%3A-50)&outputCrs=EPSG%3A31975&format=image%2Fpng&maxResults=1&label=Foo,baR"
```
For EDR (GET):
```
curl -Ln -bj "http://localhost:3000/ogc-api-edr/1.1.0/collections/C1234208438-POCLOUD/area?maxResults=1&parameter-name=all&coords=POLYGON%20%28%28-164.8%2032.3%2C%20-179.3%2025.2%2C%20-165.5%2018.3%2C%20-164.8%2032.3%29%29&labels=abc,123"
```
For EDR (POST):
```
curl --request POST \
        --url http://localhost:3000/ogc-api-edr/1.1.0/collections/C1233800302-EEDTEST/position \
        --header "Authorization: Bearer $MY_UAT_TOKEN" \
        --header 'Content-Type: application/json' \
        --data '{
    "coords": "POINT(-40 10)",
    "granuleId": ["G1233800343-EEDTEST"],
    "parameter-name": [
      "all"
    ],
    "crs": "EPSG:31975",
    "f": "image/png", "label": ["abc😡123","Fu#bar"]
  }'
```
4. Check the database tables `labels` and `job_labels` and make sure you see rows for the labels you set.
Here are a couple of useful queries (change the labels to whatever you used):
```
select "j"."jobID" from jobs j JOIN jobs_labels l ON "j"."jobID" = l.job_id JOIN labels ul ON l.label_id = ul.id WHERE ul.value = 'fizz';
select "j"."jobID" from jobs j JOIN jobs_labels l ON "j"."jobID" = l.job_id JOIN labels ul ON l.label_id = ul.id WHERE ul.value = 'abc😡123';
```


## PR Acceptance Checklist
